client.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/mendersoftware/log"
+)
+
+var (
+	errorLoadingClientCertificate = errors.New("Failed to load certificate and key")
```

Comment:
I would use capital letter, in main import/export for method does not matter (you can't import main package to another project). otherwise you almost always want to export predefined errors in stand alone package so package user can compare errors